catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/model/SearchTest.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.ui.searchui.query.model;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import ddf.action.Action;
+import ddf.action.ActionRegistry;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SearchTest {
+
+    private static final String[] SOURCE_ID_VALUES = {"ryan", "lombardi", "john", "armand"};
+
+    private static final String EXCEPTION_MESSAGE = "This is the message for id = [%s]";
+
+    private Search searchProvider;
+
+    private SearchRequest searchRequest;
+
+    @Before
+    public void setup() throws Exception {
+        searchRequest = mock(SearchRequest.class);
+        Set<String> sourceIds = new HashSet<>(Arrays.asList(SOURCE_ID_VALUES));
+        when(searchRequest.getSourceIds()).thenReturn(sourceIds);
+
+        searchProvider = new Search(searchRequest, new ActionRegistry() {
+            @Override
+            public <T> List<Action> list(T subject) {
+                return null;
+            }
+        });
+    }
+
+    @Test
+    public void testFailSource() throws Exception {
```

Comment:
After further analysis of this code, the combinatorics of different sources are not meant to be tested here. The same code is run, regardless of the source. The base cases are covered here, in `SearchTest` - and I have a test for each major exception (along with recursion cases) in `SearchControllerTest` which I believe to be sufficient. 

Please let me know if there is a crack (or gaping hole) in my testing strategy.